backend/courses/util.py

```diff
@@ -712,3 +712,46 @@ def get_semesters(semesters: str = None) -> list[str]:
             if s not in possible_semesters:
                 raise ValueError(f"Provided semester {s} was not found in the db.")
     return sorted(semesters)
+
+
+def historical_semester_probability(current, courses):
+    """
+    :param current: The current semester represented in the 20XX(A|B|C) format.
+    :type current: str
+    :param courses: A list of Course objects sorted by date in ascending order.
+    :type courses: list
+    :returns: A list of 3 probabilities representing the likelihood of
+    taking a course in each semester.
+    :rtype: list
+    """
+    prob_distribution = [0.4, 0.3, 0.15, 0.1, 0.05]
+
+    def normalize_and_round(prob, i):
+        """Modifies the probability distribution to account for the
+        fact that the last course was taken i semesters ago."""
+        truncate = prob_distribution[:i]
+        total = sum(truncate)
+        return list(map(lambda x: round(x / total, 3), truncate))
+
+    current_index = int(translate_semester(current)) // 10
+    min_index = current_index - 60
+    max_index = current_index - 10
+    p = [0, 0, 0]
```

Comment:
is it fine that we return an invalid probability distribution in this case?

Reply:
The think the idea is that [A,B,C] is not a probability distribution but instead each individual element represents a probability (if the class occurs for the past 5 years for the spring for example, then p[0] = 1)